Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -5463,8 +6063,10 @@ class BBQJIT {
             BLOCK(
                 if (m_jit.supportsCountPopulation())
                     m_jit.countPopulation64(operandLocation.asGPR(), resultLocation.asGPR());
-                else
-                    emitCCall(&operationPopcount64, Vector<Value> { operand }, TypeKind::I32, result);
+                else {
+                    result = topValue(TypeKind::I32);
```

Comment:
This isn't strictly related to your patch but I just realized this is slightly wrong...`operationPopcount64` returns a 64-bit result, not a 32-bit one, so `I32` is the wrong type. If we reuse the `result` from `EMIT_UNARY` like I mentioned in my previous comment, it should have the correct type already; otherwise, this should probably use `TypeKind::I64`.

Reply:
Fixed via switching it to `consume`.